### PR TITLE
GraphQL Voyager

### PR DESCRIPTION
**Pull request by AndrewOlenik**
GraphQL Voyager #23

### CODE REVIEW

File: api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.DispatcherServlet;
+
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnClass(DispatcherServlet.class)
+public class VoyagerConfiguration {
+
+    @Bean
+    VoyagerController voyagerController() {
```
Comment (trioletas) [inline api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerConfiguration.java]: Can we make it configurable just like graphiql?
```java
    @ConditionalOnProperty(value = "voyager.enabled", havingValue = "true", matchIfMissing = true)
```

File: api/src/main/resources/voyager.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+<div id="voyager">Loading...</div>
+<script>
+    // Defines a GraphQL introspection fetcher using the fetch API. You're not required to
+    // use fetch, and could instead implement introspectionProvider however you like,
+    // as long as it returns a Promise
+    // Voyager passes introspectionQuery as an agrument for this function
+    function introspectionProvider(introspectionQuery) {
+        // This example expects a GraphQL server at the path /graphql.
+        // Change this to point wherever you host your GraphQL server.
+        return fetch('/graphql', {
```
Comment (trioletas) [inline api/src/main/resources/voyager.html]: graphql endpoint should be configurable just like in [graphiql-spring-boot-autoconfigure](https://github.com/graphql-java/graphql-spring-boot/blob/master/graphiql-spring-boot-autoconfigure/src/main/resources/graphiql.html#L87)

File: api/src/main/resources/voyager.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <style>
```
Comment (trioletas) [inline api/src/main/resources/voyager.html]: please add

1) `title` and ability to configure it [externally](https://github.com/graphql-java/graphql-spring-boot/blob/master/graphiql-spring-boot-autoconfigure/src/main/java/com/oembedler/moon/graphiql/boot/GraphiQLController.java#L25):
` <title>${pageTitle}</title>`

2) meta tags from [graphql-voyager node.js middleware](https://github.com/APIs-guru/graphql-voyager/blob/master/src/middleware/render-voyager-page.ts#L17):
```html
 <meta charset=utf-8 />
 <meta name="viewport" content="user-scalable=no, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, minimal-ui">
```

File: api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Controller
+public class VoyagerController {
+
+    @RequestMapping(value = "/voyager")
```
Comment (trioletas) [inline api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerController.java]: should make it configurable (with default value) just like in [graphiql-spring-boot-autoconfigure](https://github.com/graphql-java/graphql-spring-boot/blob/master/graphiql-spring-boot-autoconfigure/src/main/java/com/oembedler/moon/graphiql/boot/GraphiQLController.java#L28)

Comment (AndrewOlenik) [reply api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerController.java]: Ok, I see. Shall we consider extract it to voyager-spring-boot-autoconfigure?

Comment (trioletas) [reply api/src/main/java/lv/ctco/tpl/bff/voyager/VoyagerController.java]: Eventually, yes but for now let’s keep it a) simple and b) in the template

**Comment by trioletas** [comment]
Since graphql voyager integration is pretty unique to our template the only thing we’re missing to merge this thing is some docs on how to configure the voyager specific things, please add those to readme and we’re good to merge

**Pull request merged**